### ISSUE TITLE
Chapter 2: Stratification using train_test_split()?

### ISSUE THREAD

**Issue by EricThomson** [ageron/handson-ml2#81]
In Chapter 2, instead of using `StratifiedShuffleSplit()` (which is relatively complicated), is there any harm to just sticking with our old stand-by `train_test_split()`, which has a `stratify` keyword?

    strat_test_set, strat_train_set =     
    train_test_split(housing, test_size = 0.2, stratify = housing['income_cat'])

**Comment by AntoineToussaint** [comment]
I had the same question. Seems to achieve the same result.

**Comment by lord8266** [comment]
`train_test_split()` with stratify is a more recent addition which might have not been available during the making of this notebook

Nevertheless  `StratifiedShuffleSplit()` can generate multiple splits whereas `train_test_split()` can do only one

**Comment by AntoineToussaint** [comment]
Thanks very much for the answer.

**Comment by ageron** [comment]
Thanks @EricThomson for your question and @lord8266 for your answer. 
The `train_test_split()` function is simple to use, you can definitely go ahead and use it. The `StratifiedShuffleSplit()` solution is useful if you want to perform stratified sampling, which makes sense when you want to guarantee that the classes are represented in the same proportions in the training set and the test set (in the book, I use it to ensure that there's the same proportion of people of various income groups in the training set and the test set). In general, you won't need to do this, it was meant as an example of more advanced splitting functions that you may want to use.

I'll close this issue, but if it's still unclear, please feel free to reopen.

Hope this helps!
